src/transformers/modeling_auto.py

```diff
@@ -620,6 +662,10 @@ def from_config(cls, config):
             config = BertConfig.from_pretrained('bert-base-uncased')    # Download configuration from S3 and cache.
             model = AutoModelWithLMHead.from_config(config)  # E.g. model was saved using `save_pretrained('./test/saved_model/')`
         """
+        warnings.warn(
+            "The class `AutoModelWithLMHead` is deprecated and will be removed in a future version. Please use `AutoModelForCausalLM` for causal language models, `AutoModelForMaskedLM` for masked language models and `AutoModelForSeq2SeqLM` for encoder-decoder models.",
+            DeprecationWarning,
```

Comment:
I think it should be a `FutureWarning` rather than a `DeprecationWarning` so that users are aware of this change (`DeprecationWarnings` are ignored by default)